packages/components/src/templates/next/components/internal/SearchableTable/SearchableTableClient.tsx

```diff
@@ -0,0 +1,221 @@
+import { CellContent } from "./CellContent"
+import { MAX_NUMBER_OF_COLUMNS, PAGINATION_MAX_ITEMS } from "./constants"
+import { getFilteredItems } from "./getFilteredItems"
+import { getPaginatedItems } from "./getPaginatedItems"
+
+const createSearchableTableStyles = tv({
+  slots: {
+    container: "mx-auto w-full",
+    title: "prose-display-md break-words text-base-content-strong",
+    search: "mt-9",
+    tableContainer: "mt-8 overflow-x-auto",
+    table:
+      "[&_>_tbody_>_tr:nth-child(even)_>_td]:bg-base-canvas-default w-full border-collapse border-spacing-0 [&_>_tbody_>_tr:nth-child(odd)_>_td]:bg-base-canvas-alt",
+    tableRow: "text-left",
+    tableCell:
+      "max-w-40 break-words border border-base-divider-medium px-4 py-3 align-top last:max-w-full [&_li]:my-0 [&_li]:pl-1 [&_ol]:mt-0 [&_ol]:ps-5 [&_ul]:mt-0 [&_ul]:ps-5",
+    emptyState:
+      "flex flex-col items-center justify-center gap-8 self-stretch px-10 py-20 pt-24",
+    emptyStateHeadings: "text-center",
+    emptyStateTitle: "prose-headline-lg-regular",
+    emptyStateSubtitle:
+      "text-base-content-default prose-headline-lg-regular mt-3",
+    emptyStateButton:
+      "prose-headline-base-medium text-link hover:text-link-hover",
+    pagination: "mt-8 flex w-full justify-center lg:justify-end",
+  },
+  variants: {
+    isHeader: {
+      true: {
+        tableCell:
+          "bg-brand-interaction text-base-content-inverse [&_ol]:prose-label-md-medium [&_p]:prose-label-md-medium",
+      },
+      false: {
+        tableCell: "text-base-content [&_ol]:prose-body-sm [&_p]:prose-body-sm",
+      },
+    },
+    bold: {
+      true: {
+        emptyStateTitle: "text-base-content-strong",
+      },
+      false: {
+        emptyStateTitle: "text-base-content-subtle",
+      },
+    },
+  },
+})
+
+const compoundStyles = createSearchableTableStyles()
+
+export type SearchableTableClientProps = Omit<SearchableTableProps, "items"> & {
+  items: {
+    row: SearchableTableProps["items"][number]
+    key: string
+  }[]
+}
+
+export const SearchableTableClient = ({
+  title,
+  headers,
+  items,
+  sitemap,
+  LinkComponent,
+}: SearchableTableClientProps) => {
+  const [_search, setSearch] = useState("")
+  const search = useDeferredValue(_search)
+  const [currPage, setCurrPage] = useState(1)
+  const titleId = useId()
+
+  const maxNoOfColumns = Math.min(
+    headers.length,
+    ...items.map((row) => row.row.length),
+    MAX_NUMBER_OF_COLUMNS,
+  )
+  const filteredItems = useMemo(
+    () =>
+      getFilteredItems({
+        items,
+        searchValue: search,
+      }),
+    [items, search],
+  )
+
+  const paginatedItems = useMemo(
+    () =>
+      getPaginatedItems({
+        items: filteredItems,
+        currPage,
+        itemsPerPage: PAGINATION_MAX_ITEMS,
+      }),
+    [currPage, filteredItems],
+  )
+
+  const isInitiallyEmpty = items.length === 0 || maxNoOfColumns === 0
+  const isFilteredEmpty = items.length !== 0 && filteredItems.length === 0
```

Comment:
nit: should we be using `!!search` rather than `items.length`?

Reply:
Hmm but if the table is initially empty, then using the search value means the user types in a search value will display as "no results found" but actually there is no results at the beginning.

Was thinking of sticking to displaying the message that there is no results in the beginning, regardless of the inputs by the user.

Reply:
yea i think both are fine! was just thinking that having a search term means that we shuold reflect that, regardless of collection state. this is very minor though, happy to take either